nss-day-cohort-45/tabloidfullstack-giving-gillyflowers
# Upload Header Image

### Changes:
- added `UploadController` to handle POST calls to `/api/upload`
- added `uploadFile` method to `PostProvider.js`
- added affordance for uploading images to `PostForm.js` with a select element toggling between uploading an image or supplying a URL
---
### Testing:
- fetch branch, run app, API, and log in
- after fetching the branch, create a directory called `uploads` in `Tabloid/client/public` for the images to be saved to (if it doesn't already exist)
- click New Post in the navbar to visit the Post Form component
- fill out all the fields and select Upload Image from the dropdown
- choose an image file (.png, .jpg, .gif, .bmp) from your computer and hit Upload Image to store it on the server and have the image path saved to your new post object
- hit Submit to save your post
- select your new post from the list of posts and see that your header image shows up
